### DISCUSSION TITLE
Defining pagination

### DISCUSSION THREAD

**Discussion by ctron** [guacsec/trustify#91] [Ideas]
Coming from a discussion on https://github.com/trustification/trustify/pull/82 … I think we need to define how "pagination" works.

As a goal, I think it makes sense to leverage the same logic/struct of pagination for all APIs. Re-using code as good as possible. But not forcing pagination on all APIs. Only if it makes sense.

While we have a current definition of pagination on the backend, I personally thinks it's a bit verbose. Then again, I don't have a strong opinion other than: please let's re-use the same logic for all APIs.

I think the minimum information we need is:

* Input: `limit`/`offset`, or `page_size`/`page`.
* SeaORM uses `page_size`/`page` out of the box, but I am sure we can implement other patterns as well
* Output: `number_of_items` or `page_size`/`number_of_pages`.

Of course we can add more. 

@carlosthe19916 As I think the UI is the most important component working with this, what's your preference?

**Comment by carlosthe19916** [comment]
> As a goal, I think it makes sense to leverage the same logic/struct of pagination for all APIs. Re-using code as good as possible. But not forcing pagination on all APIs. Only if it makes sense.

Fully agree. Pagination only when it makes sense

The UI only needs:
- Input: `limit` & `offset`
- Output: `number_of_items | total`  plus the array of items that match the query criteria.

## API language
IMHO, The REST API should talk in its own "language". It means, to use `offset`, `limit`, `total | number_of_items`

## UI Language
The UI takes the REST API "language" and translates it to human "language" so the UI will translates `offset` & `limit` to `page` & `page size`.

This way both UI and API are clearly separated

**Reply by ctron** [reply]
Ok, I'll come up with a PR on top of PR #82 and we can take a look then.

**Reply by ctron** [reply]
Here is the PR: https://github.com/trustification/trustify/pull/92

**Reply by bobmcwhirter** [reply]
I left a comment but defer full review to @jcrossley3 and @carlosthe19916 who hold opinions.

**Reply by ctron** [reply]
> I left a comment but defer full review to @jcrossley3 and @carlosthe19916 who hold opinions.

 I am not sure I can see it.

**Reply by bobmcwhirter** [reply]
It was just the `PhantomData<...>` ish.

**Comment by bobmcwhirter** [comment]
Just tidying up, do we consider this discussion "resolved"?
